python/audit/utils.py

```diff
@@ -86,6 +91,70 @@ def _get_deleted_obj(view: View, request: Request, kwargs) -> Model | None:
     return deleted_obj
 
 
+def _get_object_changes(prev_data: dict, current_obj: Model) -> dict:
+    serializer_class = None
+    if isinstance(current_obj, Group):
+        serializer_class = GroupAuditSerializer
+    elif isinstance(current_obj, Role):
+        serializer_class = RoleAuditSerializer
+    elif isinstance(current_obj, User):
+        serializer_class = UserAuditSerializer
+    elif isinstance(current_obj, Policy):
+        serializer_class = PolicyAuditSerializer
+
+    if not serializer_class:
+        return {}
+
+    current_data = serializer_class(current_obj).data
+    current_fields = {k: v for k, v in current_data.items() if prev_data[k] != v}
+    if not current_fields:
+        return current_fields
+
+    object_changes = {
+        "current": current_fields,
+        "previous": {k: v for k, v in prev_data.items() if k in current_fields},
+    }
+    if object_changes["current"].get("password"):
```

Comment:
`.get()` by default return `None`, if key is not found. Maybe use `if object_changes["current"].get("password") is not None:` ?